Error: Cannot find module 'nx/package.json'

### Current Behavior

Hi,

We are using NX on our TeamCity build server. It seems that the latest version, 15.8.9, is not working for us and throws the following error when running the Cypress end-to-end testing (or another NX command, e.g. nx report) on our Angular project. Note we add NX as a global package:

_Error: Cannot find module 'nx/package.json'_

![image](https://user-images.githubusercontent.com/67626158/228110319-2e15db5f-7f89-489e-a3b6-b905beeeca28.png)


Doing precisely the same, but downgrading NX to version 15.8.7 works.

![image](https://user-images.githubusercontent.com/67626158/228113422-b55d756e-6cb4-415c-85d9-2b8a27cc8ff4.png)

Also, we can't retrieve the broken version report since it does not allow us to run NX commands. I've attached a report for 15.8.7 (which is working). 

Thanks for your help.

Regards,

### Expected Behavior

Expect NX 15.8.8-9 to work the same as 15.8.7

### GitHub Repo

_No response_

### Steps to Reproduce

1. Install NX globally 15.8.9 `yarn global add nx@15.8.9` 
2. Run `yarn install` on NX Angular project
3. Run Cypress e2e (or another nx cli command)


### Nx Report

```shell
Node : 16.18.1
OS   : linux x64
yarn : 1.22.19

nx : Not Found
@nrwl/angular : 12.6.3
@nrwl/cli : 12.6.3
@nrwl/cypress : 12.6.3
@nrwl/devkit : 12.6.3
@nrwl/eslint-plugin-nx : Not Found
@nrwl/express : 12.6.3
@nrwl/jest : 12.6.3
@nrwl/linter : 12.6.3
@nrwl/nest : Not Found
@nrwl/next : Not Found
@nrwl/node : 12.6.3
@nrwl/nx-cloud : Not Found
@nrwl/react : Not Found
@nrwl/schematics : Not Found
@nrwl/tao : 12.6.3
@nrwl/web : Not Found
@nrwl/workspace : 12.6.3
@nrwl/storybook : 12.6.3
@nrwl/gatsby : Not Found
typescript : 4.3.5
```


### Failure Logs

```shell
[08:06:28] :	 [Step 3/16] $ node ./decorate-angular-cli.js && ngcc --properties es2015 browser module main
[08:06:29] :	 [Step 3/16] 
[08:06:29] :	 [Step 3/16] >  NX  Angular CLI has been decorated to enable computation caching.
[08:06:29] :	 [Step 3/16] 
[08:06:38] :	 [Step 3/16] Compiling @angular/animations : es2015 as esm2015
[08:06:38] :	 [Step 3/16] Compiling @angular/core : es2015 as esm2015
[08:06:38] :	 [Step 3/16] Compiling @angular/compiler/testing : es2015 as esm2015
[08:06:39] :	 [Step 3/16] Compiling @nrwl/angular/testing : es2015 as esm2015
[08:06:39] :	 [Step 3/16] Compiling @angular/animations : main as umd
[08:06:40] :	 [Step 3/16] Compiling @angular/compiler/testing : main as umd
[08:06:40] :	 [Step 3/16] Compiling @nrwl/angular/testing : main as umd
[08:06:42] :	 [Step 3/16] Compiling @angular/common : es2015 as esm2015
[08:06:42] :	 [Step 3/16] Compiling @angular/core/testing : es2015 as esm2015
[08:06:42] :	 [Step 3/16] Compiling @angular/animations/browser : es2015 as esm2015
[08:06:43] :	 [Step 3/16] Compiling @ngrx/store : es2015 as esm2015
[08:06:43] :	 [Step 3/16] Compiling @angular/platform-browser : es2015 as esm2015
[08:06:43] :	 [Step 3/16] Compiling @angular/common/http : es2015 as esm2015
[08:06:44] :	 [Step 3/16] Compiling @angular/forms : es2015 as esm2015
[08:06:44] :	 [Step 3/16] Compiling @angular/platform-browser-dynamic : es2015 as esm2015
[08:06:44] :	 [Step 3/16] Compiling @angular/platform-browser/animations : es2015 as esm2015
[08:06:44] :	 [Step 3/16] Compiling @angular/platform-browser/testing : es2015 as esm2015
[08:06:44] :	 [Step 3/16] Compiling @angular/platform-browser-dynamic/testing : es2015 as esm2015
[08:06:44] :	 [Step 3/16] Compiling @angular/platform-server : es2015 as esm2015
[08:06:45] :	 [Step 3/16] Compiling @angular/common/testing : es2015 as esm2015
[08:06:45] :	 [Step 3/16] Compiling @angular/router : es2015 as esm2015
[08:06:45] :	 [Step 3/16] Compiling @fortawesome/angular-fontawesome : es2015 as esm2015
[08:06:45] :	 [Step 3/16] Compiling @nguniversal/common/engine : es2015 as esm2015
[08:06:45] :	 [Step 3/16] Compiling @nguniversal/express-engine/tokens : es2015 as esm2015
[08:06:45] :	 [Step 3/16] Compiling @ng-bootstrap/ng-bootstrap : es2015 as esm2015
[08:06:46] :	 [Step 3/16] Compiling @ng-select/ng-select : es2015 as esm2015
[08:06:47] :	 [Step 3/16] Compiling ngx-cookie-service : es2015 as esm2015
[08:06:47] :	 [Step 3/16] Compiling ngx-ellipsis : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling ngx-page-scroll-core : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @angular/animations/browser/testing : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @angular/common/http/testing : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @angular/elements : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @angular/platform-server/init : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @angular/platform-server/testing : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @angular/router/testing : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @fortawesome/angular-fontawesome/testing : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @ngrx/store/testing : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @nguniversal/common : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @nguniversal/express-engine : es2015 as esm2015
[08:06:48] :	 [Step 3/16] Compiling @nguniversal/common/clover : es2015 as esm2015
[08:06:49] :	 [Step 3/16] Compiling @ngx-ssr/timeout : es2015 as esm2015
[08:06:49] :	 [Step 3/16] Compiling ngx-nested-ellipsis : es2015 as esm2015
[08:06:49] :	 [Step 3/16] Compiling @angular/animations/browser : main as umd
[08:06:49] :	 [Step 3/16] Compiling @angular/animations/browser/testing : main as umd
[08:06:49] :	 [Step 3/16] Compiling @angular/core : main as umd
[08:06:50] :	 [Step 3/16] Compiling @angular/common : main as umd
[08:06:50] :	 [Step 3/16] Compiling @angular/common/http : main as umd
[08:06:50] :	 [Step 3/16] Compiling @angular/common/http/testing : main as umd
[08:06:50] :	 [Step 3/16] Compiling @angular/elements : main as umd
[08:06:50] :	 [Step 3/16] Compiling @angular/platform-browser : main as umd
[08:06:50] :	 [Step 3/16] Compiling @angular/platform-server/init : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/core/testing : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/platform-browser-dynamic : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/platform-browser/testing : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/platform-browser-dynamic/testing : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/platform-browser/animations : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/platform-server : main as umd
[08:06:51] :	 [Step 3/16] Compiling @angular/platform-server/testing : main as umd
[08:06:52] :	 [Step 3/16] Compiling @angular/common/testing : main as umd
[08:06:52] :	 [Step 3/16] Compiling @angular/router/testing : main as umd
[08:06:52] :	 [Step 3/16] Compiling @angular/router : main as umd
[08:06:52] :	 [Step 3/16] Compiling @fortawesome/angular-fontawesome : main as umd
[08:06:52] :	 [Step 3/16] Compiling @fortawesome/angular-fontawesome/testing : main as umd
[08:06:52] :	 [Step 3/16] Compiling @ng-bootstrap/ng-bootstrap : module as esm5
[08:06:52] :	 [Step 3/16] Compiling @angular/forms : main as umd
[08:06:53] :	 [Step 3/16] Compiling @ng-bootstrap/ng-bootstrap : main as umd
[08:06:53] :	 [Step 3/16] Compiling @ng-select/ng-select : module as esm5
[08:06:54] :	 [Step 3/16] Compiling @ng-select/ng-select : main as umd
[08:06:54] :	 [Step 3/16] Compiling @ngrx/store : main as umd
[08:06:54] :	 [Step 3/16] Compiling @ngrx/store/testing : main as umd
[08:06:54] :	 [Step 3/16] Compiling @nguniversal/common : main as umd
[08:06:54] :	 [Step 3/16] Compiling @nguniversal/common/clover : main as umd
[08:06:55] :	 [Step 3/16] Compiling @nguniversal/express-engine/tokens : main as umd
[08:06:55] :	 [Step 3/16] Compiling @nguniversal/express-engine : main as umd
[08:06:55] :	 [Step 3/16] Compiling @nguniversal/common/engine : main as umd
[08:06:55] :	 [Step 3/16] Compiling @ngx-ssr/timeout : main as umd
[08:06:55] :	 [Step 3/16] Compiling ngx-cookie-service : module as esm5
[08:06:55] :	 [Step 3/16] Compiling ngx-cookie-service : main as umd
[08:06:55] :	 [Step 3/16] Compiling ngx-ellipsis : module as esm5
[08:06:55] :	 [Step 3/16] Compiling ngx-ellipsis : main as umd
[08:06:55] :	 [Step 3/16] Compiling ngx-embed-video : main as commonjs
[08:06:56] :	 [Step 3/16] Compiling ngx-nested-ellipsis : main as umd
[08:06:56] :	 [Step 3/16] Compiling ngx-page-scroll-core : module as esm5
[08:06:56] :	 [Step 3/16] Compiling ngx-page-scroll-core : main as umd
[08:06:57] :	 [Step 3/16] Done in 398.44s.
[08:06:57]W:	 [Step 3/16] node:internal/modules/cjs/loader:988
[08:06:57]W:	 [Step 3/16]   throw err;
[08:06:57]W:	 [Step 3/16]   ^
[08:06:57]W:	 [Step 3/16] 
[08:06:57]W:	 [Step 3/16] Error: Cannot find module 'nx/package.json'
[08:06:57]W:	 [Step 3/16] Require stack:
[08:06:57]W:	 [Step 3/16] - /usr/local/share/.config/yarn/global/node_modules/nx/bin/nx.js
[08:06:57]W:	 [Step 3/16]     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:985:15)
[08:06:57]W:	 [Step 3/16]     at Function.resolve (node:internal/modules/cjs/helpers:109:19)
[08:06:57]W:	 [Step 3/16]     at getLocalNxVersion (/usr/local/share/.config/yarn/global/node_modules/nx/bin/nx.js:125:50)
[08:06:57]W:	 [Step 3/16]     at Object.<anonymous> (/usr/local/share/.config/yarn/global/node_modules/nx/bin/nx.js:68:42)
[08:06:57]W:	 [Step 3/16]     at Module._compile (node:internal/modules/cjs/loader:1155:14)
[08:06:57]W:	 [Step 3/16]     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
[08:06:57]W:	 [Step 3/16]     at Module.load (node:internal/modules/cjs/loader:1033:32)
[08:06:57]W:	 [Step 3/16]     at Function.Module._load (node:internal/modules/cjs/loader:868:12)
[08:06:57]W:	 [Step 3/16]     at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
[08:06:57]W:	 [Step 3/16]     at node:internal/main/run_main_module:22:47 {
[08:06:57]W:	 [Step 3/16]   code: 'MODULE_NOT_FOUND',
[08:06:57]W:	 [Step 3/16]   requireStack: [ '/usr/local/share/.config/yarn/global/node_modules/nx/bin/nx.js' ]
[08:06:57]W:	 [Step 3/16] }
[08:06:57]i:	 [Step 3/16] Docker event: {"status":"die","id":"afd7b20242962a6c1a3d3837127bffb15b146a27c7800a5f0981f4e914786722","from":"cypress/browsers:node-16.18.1-chrome-110.0.5481.96-1-ff-109.0-edge-110.0.1587.41-1","Type":"container","Action":"die","Actor":{"ID":"afd7b20242962a6c1a3d3837127bffb15b146a27c7800a5f0981f4e914786722","Attributes":{"exitCode":"1","image":"cypress/browsers:node-16.18.1-chrome-110.0.5481.96-1-ff-109.0-edge-110.0.1587.41-1","jetbrains.teamcity.buildId":"22837","name":"vibrant_brahmagupta"}},"scope":"local","time":1679962017,"timeNano":1679962017615085957}
[08:07:07]i:	 [Step 3/16] Docker event: {"status":"destroy","id":"afd7b20242962a6c1a3d3837127bffb15b146a27c7800a5f0981f4e914786722","from":"cypress/browsers:node-16.18.1-chrome-110.0.5481.96-1-ff-109.0-edge-110.0.1587.41-1","Type":"container","Action":"destroy","Actor":{"ID":"afd7b20242962a6c1a3d3837127bffb15b146a27c7800a5f0981f4e914786722","Attributes":{"image":"cypress/browsers:node-16.18.1-chrome-110.0.5481.96-1-ff-109.0-edge-110.0.1587.41-1","jetbrains.teamcity.buildId":"22837","name":"vibrant_brahmagupta"}},"scope":"local","time":1679962027,"timeNano":1679962027957327267}
[08:07:07]W:	 [Step 3/16] Process exited with code 1
[08:07:07]E:	 [Step 3/16] Process exited with code 1 (Step: Run Cypress UI Tests (Command Line))
[08:07:08]i:	 [Step 3/16] Docker wrapper: restore directory ownership (14s)
[08:07:08]i:		 [Docker wrapper: restore directory ownership] Set ownership to 1000/1000 for "/opt/buildagent/work/e6e79467c8b2eda7" "/opt/buildagent/temp/agentTmp" "/opt/buildagent/temp/buildTmp" "/opt/buildagent/system" 
[08:07:08]i:		 [Docker wrapper: restore directory ownership] Docker event: {"status":"create","id":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","from":"busybox","Type":"container","Action":"create","Actor":{"ID":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","Attributes":{"image":"busybox","name":"sweet_swanson"}},"scope":"local","time":1679962028,"timeNano":1679962028198891282}
[08:07:08]i:		 [Docker wrapper: restore directory ownership] Docker event: {"status":"attach","id":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","from":"busybox","Type":"container","Action":"attach","Actor":{"ID":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","Attributes":{"image":"busybox","name":"sweet_swanson"}},"scope":"local","time":1679962028,"timeNano":1679962028204376717}
[08:07:08]i:		 [Docker wrapper: restore directory ownership] Docker event: {"status":"start","id":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","from":"busybox","Type":"container","Action":"start","Actor":{"ID":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","Attributes":{"image":"busybox","name":"sweet_swanson"}},"scope":"local","time":1679962028,"timeNano":1679962028675075402}
[08:07:22]i:		 [Docker wrapper: restore directory ownership] Docker event: {"status":"die","id":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","from":"busybox","Type":"container","Action":"die","Actor":{"ID":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","Attributes":{"exitCode":"0","image":"busybox","name":"sweet_swanson"}},"scope":"local","time":1679962042,"timeNano":1679962042494032375}
[08:07:22]i:		 [Docker wrapper: restore directory ownership] Docker event: {"status":"destroy","id":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","from":"busybox","Type":"container","Action":"destroy","Actor":{"ID":"fdd64ae6ca74a8f71b79e413a2f73a4cd3ee135cade9f6010aa66db66340581a","Attributes":{"image":"busybox","name":"sweet_swanson"}},"scope":"local","time":1679962042,"timeNano":1679962042800549058}
[08:07:22]E:	 [Step 3/16] Step Run Cypress UI Tests (Command Line) failed
```


### Additional Information

_No response_

--- a/packages/nx/bin/nx.ts
+++ b/packages/nx/bin/nx.ts
@@ -11,8 +11,8 @@ import {
   getNxRequirePaths,
 } from '../src/utils/installation-directory';
 import { major } from 'semver';
-import { readJsonFile } from '../src/utils/fileutils';
 import { stripIndents } from '../src/utils/strip-indents';
+import { readModulePackageJson } from '../src/utils/package-json';
 import { execSync } from 'child_process';
 
 function main() {
@@ -163,12 +163,17 @@ function warnIfUsingOutdatedGlobalInstall(
   }
 }
 
-function getLocalNxVersion(workspace: WorkspaceTypeAndRoot): string {
-  return readJsonFile(
-    require.resolve('nx/package.json', {
-      paths: getNxRequirePaths(workspace.dir),
-    })
-  ).version;
+function getLocalNxVersion(workspace: WorkspaceTypeAndRoot): string | null {
+  const localNxPackages = ['nx', '@nrwl/tao', '@nrwl/cli'];
+  for (const pkg of localNxPackages) {
+    try {
+      const { packageJson } = readModulePackageJson(
+        'nx',
+        getNxRequirePaths(workspace.dir)
+      );
+      return packageJson.version;
+    } catch {}
+  }
 }
 
 function _getLatestVersionOfNx(): string {

